lib/gc/ExecutionEngine/CPURuntime/ConstantCache.cpp

```diff
@@ -0,0 +1,40 @@
+#include "gc/ExecutionEngine/CPURuntime/ConstantCache.hpp"
+#include <string.h>
+#include <unordered_map>
+
+namespace mlir::gc {
+
+const_cache_proxy::~const_cache_proxy() = default;
+
+
+cached_graph_tensor::cached_graph_tensor(
+    const std::shared_ptr<const_cache_proxy> &base, size_t offset)
+    : base{base}, offset{offset} {
+  // todo: fill in real values
+  ref.basePtr = (char *)base->get_buffer_unsafe() + offset;
+  ref.data = ref.basePtr;
+  ref.offset = 0;
+  memset(ref.sizes, 0, sizeof(ref.sizes));
+  memset(ref.strides, 0, sizeof(ref.strides));
+}
+
+static std::unordered_map<uint64_t, std::shared_ptr<cached_graph_tensor>> cache;
+
+std::shared_ptr<cached_graph_tensor> query_cached_tensor(uint64_t key) {
```

Comment:
This is a dummy interface of cache manager, just for testing when const cache is not available. @niuxiaog Feel free to remove/modify the cache manager interfaces here.

Reply:
My implementation is at https://github.com/intel/graph-compiler/commit/5eb0ac014cd3fdf48a27eee05c71bb8aed719274. Let's test it when possible.